src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -278,116 +304,3125 @@ static ColliderJntSphInit D_80907014 = {
+            if (this->unk_398 >= 0x3D) {
+                this->unk_39C = 0xA;
+                this->unk_398 = 0;
+                this->unk_410.x = 0.0f;
+            }
+            break;
+        case 10:
+            this->unk_339 = 0x18;
+            Math_ApproachF(&this->unk_3A4.x, 290.0f, 0.05f, this->unk_410.x);
+            Math_ApproachF(&this->unk_3A4.y, 1130.0f, 0.05f, this->unk_410.x * 0.25f);
+            Math_ApproachF(&this->unk_3A4.z, -260.0f, 0.05f, this->unk_410.x * 1.25f);
+            if ((this->unk_398 >= 0x28) && (this->unk_398 < 0x6F)) {
+                Math_ApproachF(&globalCtx->envCtx.unk_D8, 1.0f, 1.0f, 0.02f);
+                Math_ApproachF(&this->unk_384, 10.0f, 0.1f, 0.2f);
+                Audio_PlayActorSound2(&sZelda->actor, NA_SE_EV_GOD_LIGHTBALL_2 - SFX_FLAG);
+            } else {
+                Math_ApproachZeroF(&this->unk_384, 1.0f, 0.2f);
+            }
+            if (this->unk_398 >= 0x83) {
+                Math_ApproachF(&this->unk_3B0.y, (sZelda->actor.world.pos.y + 40.0f + 10.0f) - 20.0f, 0.1f,
+                               this->unk_410.x);
+            } else {
+                Math_ApproachF(&this->unk_3B0.y, sZelda->actor.world.pos.y + 40.0f + 10.0f, 0.05f,
+                               this->unk_410.x * 0.25f);
+            }
+            Math_ApproachF(&this->unk_410.x, 1.0f, 1.0f, 0.01f);
+            if (this->unk_398 == 0xA) {
+                sZelda->unk_3C8 = 0xC;
+            }
+            if (this->unk_398 == 0x6E) {
+                sZelda->unk_3C8 = 0xD;
+            }
+            if (this->unk_398 == 0x8C) {
+                Audio_PlayActorSound2(&sZelda->actor, NA_SE_EV_HUMAN_BOUND);
+            }
+            if (this->unk_398 < 0xA0) {
+                break;
+            }
+        case 20:
+            globalCtx->nextEntranceIndex = 0x6B;
+            gSaveContext.nextCutsceneIndex = 0xFFF2;
+            globalCtx->sceneLoadFlag = 0x14;
+            globalCtx->fadeTransition = 3;
+            globalCtx->linkAgeOnLoad = 1;
+            break;
+    }
+
+    if (this->unk_39E != 0) {
+        Gameplay_CameraSetAtEyeUp(globalCtx, this->unk_39E, &this->unk_3B0, &this->unk_3A4, &this->unk_3BC);
+    }
+
+    switch (this->unk_1AC) {
+        case 0:
+            if (Animation_OnFrame(&this->skelAnime, this->unk_194)) {
+                Animation_MorphToLoop(&this->skelAnime, &object_ganon2_Anim_034278, 0.0f);
+                this->unk_1AC = 1;
+            }
+            break;
+        case 1:
+            if ((this->unk_39C < 7) && ((globalCtx->gameplayFrames & 0x1F) == 0)) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_MGANON_BREATH);
+            }
+            break;
+    }
+}
+
+void func_80902348(BossGanon2* this, GlobalContext* globalCtx) {
+    Player* player;
+    f32 temp_f2;
+    f32 temp_f12;
+    s16 i;
+    s16 j;
+    s16 phi_v0_2;
+
+    if (this->unk_316 == 0) {
+        for (i = 0; i < 2; i++) {
```

Comment:
```suggestion
        for (i = 0; i < ARRAY_COUNT(this->unk_864); i++) {
```
`unk_864` is the elements for `unk_444`